pkg/termination/termination.go

```diff
@@ -95,61 +95,76 @@ func (h *handler) run(ctx context.Context) error {
 	logger := h.log.WithValues("node", h.nodeName)
 	logger.V(1).Info("Monitoring node termination")
 
-	if err := wait.PollImmediateUntil(h.pollInterval, func() (bool, error) {
-		req, err := http.NewRequest("GET", h.pollURL.String(), nil)
-		if err != nil {
-			return false, fmt.Errorf("could not create request %q: %w", h.pollURL.String(), err)
+	if err := wait.PollUntilContextCancel(ctx, h.pollInterval, true, func(_ context.Context) (bool, error) {
```

Comment:
What happens when the context cancels on this function?

Reply:
i misread the comments for this function, i had thought it did not return the error on cancel but looking at the code it does. we need an extra `else if` to catch the cancel i think.

Reply:
i've updated this to check for a canceled context in the else so that it will continue to the rest of the function if the context was canceled.